flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/calls/IfCallGen.scala

```diff
@@ -52,19 +52,19 @@ class IfCallGen() extends CallGenerator {
     val resultCode =
       s"""
          |// --- Start code generated by ${className[IfCallGen]}
-         |${castedResultTerm1.getCode}
-         |${castedResultTerm2.getCode}
          |${operands.head.code}
          |$resultTerm = $resultDefault;
          |if (${operands.head.resultTerm}) {
          |  ${operands(1).code}
          |  if (!${operands(1).nullTerm}) {
+         |    ${castedResultTerm1.getCode}
```

Comment:
Since we have changed the type inference in [here](https://github.com/apache/flink/pull/21927/files#diff-2c7dc9ff08ba0e7271dd3f739dcd9ccce11f0d074054074f488e6d1f27df7bf3R59),  we don't need to modify int here, right?